code/modules/client/client procs.dm

```diff
@@ -324,17 +329,54 @@
 	if(byond_version < MIN_CLIENT_VERSION)		//Out of date client.
 		return null
 
-	if(!config.guests_allowed && IsGuestKey(key))
+	if(!(config.guests_allowed || config.external_auth) && IsGuestKey(key))
 		alert(src,"This server doesn't allow guest accounts to play. Please go to http://www.byond.com/ and register for a key.","Guest","OK")
 		del(src)
 		return
 
+	clients += src
+	directory[ckey] = src
+
+
+	if (LAZYLEN(config.client_blacklist_version))
+		var/client_version = "[byond_version].[byond_build]"
+		if (client_version in config.client_blacklist_version)
+			to_chat(src, "<span class='danger'><b>Your version of BYOND is explicitly blacklisted from joining this server!</b></span>")
+			to_chat(src, "Your current version: [client_version].")
+			to_chat(src, "Visit http://www.byond.com/download/ to download a different version. Try looking for a newer one, or go one lower.")
+			log_access("Failed Login: [key] [computer_id] [address] - Blacklisted BYOND version: [client_version].")
+			del(src)
+			return 0
+
+	if(IsGuestKey(key) && config.external_auth)
+		//src.real_mob = ..()
+		src.authed = FALSE
+		var/mob/abstract/unauthed/m = new()
+		m.client = src
+		return m
+		//Do auth shit
+	else
+		. = ..()
+		src.InitClient()
+		src.InitClientLate()
+
+/client/proc/InitClient(var/tmp_ckey)
```

Comment:
This should not have an argument. We should be able to use `src.ckey` for this entire proc.